### DISCUSSION TITLE
Issue building curl package with GSS-API support

### DISCUSSION THREAD

**Discussion by brianbaltz** [msys2/MINGW-packages#20985] [Q&A]
I'm not sure if this is the best place to ask but what is the best way to build the curl package with GSS-API support? The current package builds 3 variations: openssl, gnutls and winssl.  I have tried building with `--with-gssapi-includes`, `--with-gssapi-libs` and `--disable-sspi`. It seems I need to disable 'schannel' also or I get a redefinition build error.  If it does build, I get the following runtime error:
```
Mingw-w64 runtime failure:
32 bit pseudo relocation at 00007FFE6EAA72AE out of range, targeting 000000005F04ECA0, yielding the value FFFF8001F05A79EE.
```

**Comment by Biswa96** [comment]
Please provide the full change that you have done or provide the full configure command.

**Comment by brianbaltz** [comment]
Here's a diff of from the latest PKGBUILD:
```
diff --git a/mingw-w64-curl/PKGBUILD b/mingw-w64-curl/PKGBUILD
index 9ccb261e..4c39b521 100644
--- a/mingw-w64-curl/PKGBUILD
+++ b/mingw-w64-curl/PKGBUILD
@@ -94,6 +94,7 @@ prepare() {
     0002-Hack-make-relocation-work-inside-libexec-git-core-an.patch

   autoreconf -vfi
+  sed -i -e 's/-lgssapi/-lgssapi64/g' configure
 }

 do_build() {
@@ -126,7 +127,9 @@ do_build() {
   elif [ "${_variant}" = "-openssl" ]; then
     _variant_config+=("--with-default-ssl-backend=openssl")
     _variant_config+=("--with-openssl")
-    _variant_config+=("--with-schannel")
+    #_variant_config+=("--with-schannel")
+    _variant_config+=("--with-gssapi-includes=/kfw-build/src/include")
+    _variant_config+=("--with-gssapi-libs=/kfw-build/bin")
     _variant_config+=("--with-ca-bundle=${MINGW_PREFIX}/etc/ssl/certs/ca-bundle.crt")
     _variant_config+=("--with-nghttp2=${MINGW_PREFIX}")
     _variant_config+=("--without-librtmp")
@@ -145,7 +148,7 @@ do_build() {
     --without-random \
     --disable-static \
     --enable-shared \
-    --enable-sspi \
+    --disable-sspi \
     --enable-ldap \
     --enable-ldaps \
     --with-brotli \
```

I also copied the `gssapi64.dll` in the MIT Kerberos directory to `libgssapi64.dll` so autoconf could find it.

**Reply by Biswa96** [reply]
Where is the `/kfw-build/` path coming from?

**Reply by brianbaltz** [reply]
`/kfw-build` is my built MIT Kerberos folder - containing the DLLs and headers

**Reply by brianbaltz** [reply]
Attaching build log as well. 
[mingw-w64-curl-8.8.0-1-x86_64-build.log](https://github.com/msys2/MINGW-packages/files/15420761/mingw-w64-curl-8.8.0-1-x86_64-build.log)
